### PR TITLE
Update deployment docs for 0.5.0

### PR DESCRIPTION
**Pull request by QuanWanxx**
1. Update the default Deployment document, linked to new Container App template.
2. Add the document for optional deploying to Azure Function.
3. Remove old deployment ARM template.

### CODE REVIEW

File: FhirToDataLake/deploy/templates/DeployAzureContainerApp.json [hunk 1]
```diff
@@ -91,6 +91,13 @@
             "description": "Concurrent jobs exeucting in parallel."
         }
     },
+    "customizedSchemaImageReference":{
```
Comment (hongyh13) [inline FhirToDataLake/deploy/templates/DeployAzureContainerApp.json:94]: also apply these change to DeployAzureFunction?

File: FhirToDataLake/docs/Deploy-To-Function.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    | Fhir Server Url  | The URL of the FHIR server. If the baseUri has relative parts (like http://www.example.org/r4), then the relative part must be terminated with a slash, (like http://www.example.org/r4/). |
+    | Authentication  |  Whether to access the FHIR server with managed identity authentication. Set it to false if you are using an instance of the FHIR server for Azure with public access. |
+    | Fhir Version | Version of the FHIR server. Currently only R4 is supported. |
+    | Data Start | Start time stamp of the data to be exported. |
+    | Data End | Start time stamp of the data to be exported. Leave it empty if you want to periodically export data in real time.  |
+    | Container Name | A name for the Storage Account container to which Parquet files will be written. The Storage Account with autogenerated name will automatically be created during the installation. |
+    | Customized Schema Image Reference (new)  | The customized schema image reference for the image on Container Registry. Refer [TemplateManagement](https://github.com/microsoft/FHIR-Converter/blob/main/docs/TemplateManagementCLI.md) for how to manage your template images. |
```
Comment (hongyh13) [inline FhirToDataLake/docs/Deploy-To-Function.md:27]: need to add filter image reference parameter description

File: FhirToDataLake/docs/Deployment.md [hunk 3]
```diff
@@ -31,53 +31,53 @@ This solution enables you to query against the entire FHIR data with tools such
...
-    | Type Filters (new) | For data filtering use. Use along with the requiredTypes configuration. The value is a comma-separated list of FHIR queries that further restrict the results. All data of requiredTypes will be exported if not specified. Leave it blank if no filter is applied.|
-    | Package url | The build package of the agent. You need not change this. |
-    | Deploy App Insights | Whether to deploy the Application Insights. You can find logs in the deployed application insight resource. You need not change this. |
-    | App Insight Location | The location to deploy the App Insight. You need not change this. |
+    | Type Filters (new) | For data filtering use. Use along with the requiredTypes configuration. The value is a comma-separated list of FHIR queries that further restrict the results. All data of requiredTypes will be exported if not specified. Leave it blank if no filter is applied. |
+    | Image | Maximum number of replicas running for pipeline Container App. |
```
Comment (sowu880) [inline FhirToDataLake/docs/Deployment.md:62]: Items (contents or orders) does not correspond exactly to the form. (such as Location, maxInstanceCount, storageAccountType, Customized Schema Image Reference )

File: FhirToDataLake/docs/Deployment.md [hunk 4]
```diff
@@ -31,53 +31,53 @@ This solution enables you to query against the entire FHIR data with tools such
...
 
-3. Ensure to make note of the names of the _Storage Account_ and the _Azure Function App_ created during the deployment.
+3. Ensure to make note of the names of the _Storage Account_ and the _Azure Container App_ created during the deployment.
```
Comment (sowu880) [inline FhirToDataLake/docs/Deployment.md:64]: _Storage Account_ name and Container App are generated from pipeline name. Should make note of pipeline name and container name?

File: FhirToDataLake/deploy/templates/DeployAzureContainerApp.json [hunk 5]
```diff
@@ -135,7 +135,7 @@
     },
     "image": {
         "type": "string",
-        "defaultValue": "healthplatformregistry.azurecr.io/fhirtodatalaketool:v0.5.0",
+        "defaultValue": "healthplatformregistry.azurecr.io/fhirtodatalaketool:latest",
```
Comment (QuanWanxx) [inline FhirToDataLake/deploy/templates/DeployAzureContainerApp.json:138]: @sowu880 Still use latest when we build the v0.5.0 image on healthplatform Container Registry

File: FhirToDataLake/deploy/templates/DeployAzureContainerApp.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (QuanWanxx) [reply FhirToDataLake/deploy/templates/DeployAzureContainerApp.json:94]: Updeted the screen shots, thanks.

**Review by sowu880 — APPROVED** [review]

**Pull request merged**